pydarn/plotting/rtp.py

```diff
@@ -0,0 +1,988 @@
+            # record that has that parameter
+            index_first_match = next(i for i, d in enumerate(dmap_data)
+                                     if parameter in d)
+            if isinstance(dmap_data[index_first_match][parameter], DmapArray) or\
+               isinstance(dmap_data[index_first_match][parameter], DmapScalar):
+                dmap_data = dmap2dict(dmap_data)
+        except StopIteration:
+            raise rtp_exceptions.RTPUnknownParameter(parameter)
+        cls.dmap_data = dmap_data
+        cls.__check_data_type(parameter, 'array', index_first_match)
+        start_time, end_time = cls.__determine_start_end_time(start_time,
+                                                              end_time)
+
+        # y-axis coordinates, i.e., range gates,
+        # TODO: implement variant other coordinate systems for the y-axis
+        # y shape needs to be +1 longer as requirement of how pcolormesh
+        # draws the pixels on the grid
+        y = np.arange(0, cls.dmap_data[0]['nrang']+1, 1)
+        y_max = cls.dmap_data[0]['nrang']
+
+        # z: parameter data mapped into the color mesh
+        z = np.zeros((1, y_max)) * np.nan
+
+        # x: time date data
+        x = []
+
+        # We cannot simply use numpy's built in min and max function
+        # because of the groundscatter value :(
+
+        # These flags indicate if zmin and zmax should change
+        set_zmin = True
+        set_zmax = True
+        if not zmin:
+            zmin = cls.dmap_data[index_first_match][parameter][0]
+            set_zmin = False
+        if not zmax:
+            zmax = cls.dmap_data[index_first_match][parameter][0]
+            set_zmax = False
+
+        for dmap_record in cls.dmap_data:
+            # get time difference to test if there is some gap data
+            time = cls.__time2datetime(dmap_record)
+            diff_time = 0.0
+            if time > end_time:
+                break
+            if x != []:
+                # 60.0 seconds in a minute
+                delta_diff_time = (time - x[-1])
+                diff_time = delta_diff_time.seconds/60.0
+
+            # separation roughly 2 minutes
+            if diff_time > 2.0:
+                # if there is gap data (no data recorded past 2 minutes)
+                # then fill it in with white space
+                for _ in range(0, int(np.floor(diff_time/2.0))):
+                    x.append(x[-1] + timedelta(0, 120))
+                    i = len(x) - 1  # offset since we start at 0 not 1
+                    if i > 0:
+                        z = np.insert(z, len(z), np.zeros(1, y_max) * np.nan,
+                                      axis=0)
+            # Get data for the provided beam number
+            if (beam_num == 'all' or dmap_record['bmnum'] == beam_num) and\
+               (channel == 'all' or
+                    dmap_record['channel'] == channel):
+                if start_time <= time:
```

Comment:
`time` is also a standard module, so should not be used as a variable name.

Reply:
Good catch changed to `rec_time`.